etcdserver/apply.go

```diff
@@ -59,7 +59,7 @@ type applierV3 interface {
 
 	AuthEnable() (*pb.AuthEnableResponse, error)
 	AuthDisable() (*pb.AuthDisableResponse, error)
-	Authenticate(r *pb.AuthenticateRequest) (*pb.AuthenticateResponse, error)
+	Authenticate(index uint64, r *pb.InternalAuthenticateRequest) (*pb.AuthenticateResponse, error)
```

Comment:
Authenticate (ctx context.Context, username, password string)? Then it is more generic.
 
We can put the index, token into ctx as key,value paris.